src/main/utils/pypiUtils.ts

```diff
@@ -163,24 +143,141 @@ export class PypiUtils {
+        const dependencies: Map<string, string | undefined> = isSetupPy
+            ? this.getSetupPyDirectDependencies(descriptorPath)
+            : this.getRequirementsTxtDirectDependencies(descriptorPath);
+        if (!dependencies) {
+            return pipDepTree;
+        }
+        return this.filterDependencies(dependencies, pipDepTree, false, projectName);
+    }
+
+    public static filterDependencies(
+        dependencies: Map<string, string | undefined>,
+        pipDepTree: PipDepTree[],
+        isSetupPy: boolean,
+        projectName?: string
+    ): PipDepTree[] {
+        let filtered: PipDepTree[] = [];
+        if (dependencies.size === 0) {
+            return filtered;
+        }
+        for (const dep of pipDepTree) {
+            if (dep.key === projectName) {
+                filtered.push(...this.filterDependencies(dependencies, dep.dependencies, isSetupPy));
+            }
+            if (!dependencies.has(dep.key)) {
+                continue;
+            }
+            const version: string | undefined = dependencies.get(dep.key);
+            if (version && !this.isVersionsEqual(dep, version, isSetupPy)) {
+                continue;
+            }
+            filtered.push(dep);
+        }
+        return filtered;
+    }
+
+    private static isVersionsEqual(dependencyFromPipDepTree: PipDepTree, depFromDescriptor: string, isSetupPy: boolean): boolean {
+        if (isSetupPy) {
+            return dependencyFromPipDepTree.required_version === depFromDescriptor;
+        }
+        return depFromDescriptor.endsWith(dependencyFromPipDepTree.installed_version);
+    }
+
+    private static scanForProjectName(descriptors: vscode.Uri[]): string | undefined {
+        const setupPy: vscode.Uri | undefined = descriptors.find(descriptor => descriptor.fsPath.endsWith('setup.py'));
+        if (!setupPy) {
+            return;
+        }
+        return this.searchProjectName(setupPy.fsPath);
+    }
+
+    private static getRequirementsTxtDirectDependencies(path: string): Map<string, string | undefined> {
+        const content: string = fs.readFileSync(path, 'utf8');
+        return this.matchPythonDependencies(content);
+    }
+
+    private static getSetupPyDirectDependencies(path: string): Map<string, string | undefined> {
+        const content: string = fs.readFileSync(path, 'utf8');
+        // Use a regular expression to match the install_requires field
+        let installReqRegex: RegExp = /install_requires=\[(.*)\]/gms;
+        const match: RegExpExecArray | null = installReqRegex.exec(content);
+        if (!match) {
+            return new Map<string, string | undefined>();
+        }
+        return this.matchPythonDependencies(match[1]);
+    }
+
+    private static matchPythonDependencies(rawDependencies: string) {
+        let packageRegex: RegExp = /([\w\d\-.]+)\s*(?:\[.*\])?\s*((?:\s*(?:[<>]=?|!=|===?|~=)\s*[\d\w*-.]+,?)*)/gms;
```

Comment:
Isn't the `.` already contain `\w` and `\d`?
```suggestion
        let packageRegex: RegExp = /([-.]+)\s*(?:\[.*\])?\s*((?:\s*(?:[<>]=?|!=|===?|~=)\s*[*-.]+,?)*)/gms;
```